File: Iterator/Java/ListaNumeros.java
public class ListaNumeros{
    private int numeros[];
    private int posicion;

    ListaNumeros(int x)
    {
        this.numeros = new int[x];
        this.posicion = 0;
    }

    public void add(int numero)
    {
        this.numeros[this.posicion++] = numero;
    }

    public ListaNumerosIterador iterador()
    {
        return new ListaNumerosIterador(this.numeros);
    }

}

File: Iterator/Python/ListaNumeros.py
from ListaNumerosIterador import *

class ListaNumeros:
    def __init__(self):
        self._posicion = 0
        self._numeros = []
    
    def add(self, numero):
        self._numeros.append(numero)

    def iterador(self):
        return ListaNumerosIterado(self._numeros)  

